UnicodeEncodeError when using WSDL client

I am using xsdata with a wsdl definition to make a request.

I am using this client

`from xsdata.formats.dataclass.client import Client`

and instantiating it like this

`Client.from_service(TalentPortGiveFeedback, location=ENDPOINT)`

However I get the following error when sending unicode characters, such as mandarin, in the body.

`[ERROR] UnicodeEncodeError: 'latin-1' codec can't encode characters in position 609-613: Body ('你好干得好') is not valid Latin-1. Use body.encode('utf-8') if you want to send it encoded in UTF-8.\nTraceback (most recent call last):\n  File 
...
response = client.send(request, headers={\"Authorization\": f\"Bearer {bearer_token}\"})\n  File \"/var/task/xsdata/formats/dataclass/client.py\", line 93, in send\n    response = self.transport.post(self.config.location, data=data, headers=headers)\n  File \"/var/task/xsdata/formats/dataclass/transports.py\", line 44, in post\n    res = requests.post(url, data=data, headers=headers, timeout=self.timeout)\n  File \"/var/task/requests/api.py\", line 115, in post\n    return request(\"post\", url, data=data, json=json, **kwargs)\n  File \"/var/task/requests/api.py\", line 59, in request\n    return session.request(method=method, url=url, **kwargs)\n  File \"/var/task/requests/sessions.py\", line 587, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/var/task/requests/sessions.py\", line 701, in send\n    r = adapter.send(request, **kwargs)\n  File \"/var/task/requests/adapters.py\", line 489, in send\n    resp = conn.urlopen(\n  File \"/var/task/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/var/task/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/var/task/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/var/lang/lib/python3.9/http/client.py\", line 1285, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/var/lang/lib/python3.9/http/client.py\", line 1330, in _send_request\n    body = _encode(body, 'body')\n  File \"/var/lang/lib/python3.9/http/client.py\", line 168, in _encode\n    raise UnicodeEncodeError("` 

It seems the body is being encoded to 'latin-1' but i cant find any mention of this behavior or how to override it to use 'utf-8' in your docs.

I tried setting headers for charset like this: "Content-Type": "application/xml; charset=utf-8" but to no avail.

Do you have any advice on how to proceed?

--- a/docs/examples/custom-type-mapping.rst
+++ b/docs/examples/custom-type-mapping.rst
@@ -62,8 +62,9 @@ imported objects, enables xsdata to serialize the models.
 
 
 .. testoutput::
+    :options: +NORMALIZE_WHITESPACE
 
-   <?xml version="1.0" encoding="UTF-8"?>
+    <?xml version="1.0" encoding="UTF-8"?>
     <City>
       <name>footown</name>
       <streets>

--- a/docs/json.rst
+++ b/docs/json.rst
@@ -137,6 +137,7 @@ all the imported modules to find a matching dataclass.
 .. doctest::
 
     >>> from tests.fixtures.books import *  # Import all classes
+    >>> from xsdata.formats.dataclass.parsers.config import ParserConfig
     >>> json_string = """{
     ...   "author": "Hightower, Kim",
     ...   "title": "The First Book",

--- a/docs/wsdl.rst
+++ b/docs/wsdl.rst
@@ -69,8 +69,7 @@ The client can be initialized from the an operation class directly
 
     client = Client.from_service(CalculatorSoapAdd)
     client.config
-    # Config(style='document', location='http://www.dneonline.com/calculator.asmx', transport='http://schemas.xmlsoap.org/soap/http', soap_action='http://tempuri.org/Add', input=<class 'tests.fixtures.calculator.services.CalculatorSoapAddInput'>, output=<class 'tests.fixtures.calculator.services.CalculatorSoapAddOutput'>)
-
+    # Config(style='document', location='http://www.dneonline.com/calculator.asmx', transport='http://schemas.xmlsoap.org/soap/http', soap_action='http://tempuri.org/Add', input=<class 'tests.fixtures.calculator.services.CalculatorSoapAddInput'>, output=<class 'tests.fixtures.calculator.services.CalculatorSoapAddOutput'>, encoding=None)
 
 But you can also override any properties as you see fit
 
@@ -123,3 +122,10 @@ are needed for the webservice to work can not be overwritten.,
 .. code-block::
 
     client.send(params, headers={"User-Agent": "xsdata"})
+
+
+You will need to encode the payload if you intend to send non-ascii characters.
+
+.. code-block::
+
+    client = Client.from_service(encoding="utf-8")

--- a/tests/formats/dataclass/test_client.py
+++ b/tests/formats/dataclass/test_client.py
@@ -5,6 +5,7 @@ from tests.fixtures.calculator import Add
 from tests.fixtures.calculator import CalculatorSoapAdd
 from tests.fixtures.calculator import CalculatorSoapAddInput
 from tests.fixtures.calculator import CalculatorSoapAddOutput
+from tests.fixtures.hello import HelloGetHelloAsString
 from xsdata.exceptions import ClientValueError
 from xsdata.formats.dataclass.client import Client
 from xsdata.formats.dataclass.client import Config
@@ -37,6 +38,7 @@ class ClientTests(TestCase):
             "soap_action": "http://tempuri.org/Add",
             "output": CalculatorSoapAddOutput,
             "transport": "http://schemas.xmlsoap.org/soap/http",
+            "encoding": None,
         }
 
         self.assertEqual(expected, actual)
@@ -87,6 +89,13 @@ class ClientTests(TestCase):
                 "SOAPAction": "http://tempuri.org/Add",
             },
         )
+
+    def test_prepare_payload_with_encoding(self):
+        client = Client.from_service(HelloGetHelloAsString, encoding="utf-8")
+        result = client.prepare_payload(
+            {"Body": {"getHelloAsString": {"arg0": "Χριστόδουλος"}}}
+        )
+        self.assertIn("Χριστόδουλος".encode(), result)
 
     def test_prepare_payload_raises_error_with_type_mismatch(self):
         client = Client.from_service(CalculatorSoapAdd)

--- a/xsdata/formats/dataclass/client.py
+++ b/xsdata/formats/dataclass/client.py
@@ -32,6 +32,7 @@ class Config(NamedTuple):
     soap_action: str
     input: Type
     output: Type
+    encoding: Optional[str] = None
 
     @classmethod
     def from_service(cls, obj: Any, **kwargs: Any) -> "Config":
@@ -129,4 +130,8 @@ class Client:
                 f"got `{type(obj).__name__}`"
             )
 
-        return self.serializer.render(obj)
+        result = self.serializer.render(obj)
+        if self.config.encoding:
+            return result.encode(self.config.encoding)
+
+        return result

